hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -350,6 +350,7 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
    * NOTE: DO NOT OVERRIDE THIS METHOD
    */
   override final def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
+    fileIndex.refresh()
```

Comment:
Does master branch using the new reader logic, e.g., `HoodieMergeOnReadSnapshotHadoopFsRelationFactory`, have this problem?  In the new factory, the relation factory and file index are created at the time of instantiation.  Wondering if that need to be refreshed too.

Reply:
I have reviewed the corresponding code of `HoodieMergeOnReadSnapshotHadoopFsRelationFactory`, and it appears that a refresh is still necessary at this point. This is essentially because the same Spark context reuses the relation when executing identical queries.

Reply:
This also needs to be refreshed. Since Spark will reuse relations for multiple queries on the same table, the creation during initialization is just the initial state. A refresh is required when it is actually used.